domain/src/test/java/org/oppia/domain/oppialogger/loguploader/LogUploadWorkerTest.kt

```diff
@@ -0,0 +1,245 @@
+    val workInfo = workManager.getWorkInfoById(request.id)
+
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+    assertThat(fakeEventLogger.getMostRecentEvent()).isEqualTo(eventLogTopicContext)
+  }
+
+  @Test
+  fun testWorker_logException_withoutNetwork_enqueueRequest_verifySuccess() {
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.NONE)
+    exceptionsController.logNonFatalException(exception, TEST_TIMESTAMP)
+
+    val workManager = WorkManager.getInstance(ApplicationProvider.getApplicationContext())
+
+    val inputData = Data.Builder().putString(
+      LogUploadWorker.WORKER_CASE_KEY,
+      LogUploadWorker.EXCEPTION_WORKER
+    ).build()
+
+    val request: OneTimeWorkRequest = OneTimeWorkRequestBuilder<LogUploadWorker>()
+      .setInputData(inputData)
+      .build()
+    workManager.enqueue(request)
+    testCoroutineDispatchers.runCurrent()
+    val workInfo = workManager.getWorkInfoById(request.id)
+    val exceptionGot = fakeExceptionLogger.getMostRecentException()
+
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+    assertThat(exceptionGot.message).isEqualTo("TEST")
+    assertThat(exceptionGot.stackTrace).isEqualTo(exception.stackTrace)
+    assertThat(exceptionGot.cause).isEqualTo(null)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerLogUploadWorkerTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  // TODO(#89): Move this to a common test application component.
+  @Module
+  class TestModule {
+    @Provides
+    @Singleton
+    fun provideContext(application: Application): Context {
+      return application
+    }
+
+    // TODO(#59): Either isolate these to their own shared test module, or use the real logging
+    // module in tests to avoid needing to specify these settings for tests.
+    @EnableConsoleLog
+    @Provides
+    fun provideEnableConsoleLog(): Boolean = true
+
+    @EnableFileLog
+    @Provides
+    fun provideEnableFileLog(): Boolean = false
+
+    @GlobalLogLevel
+    @Provides
+    fun provideGlobalLogLevel(): LogLevel = LogLevel.VERBOSE
+  }
+
+  @Module
+  class TestLogStorageModule {
+
+    @Provides
+    @EventLogStorageCacheSize
+    fun provideEventLogStorageCacheSize(): Int = 2
+
+    @Provides
+    @ExceptionLogStorageCacheSize
+    fun provideExceptionLogStorageSize(): Int = 2
+  }
+
+  @Module
+  interface TestLogUploaderModule {
+
+    @Binds
+    fun bindsFakeLogUploader(fakeLogUploader: FakeLogUploader): LogUploader
+  }
+
+  // TODO(#89): Move this to a common test application component.
+  @Singleton
+  @Component(
+    modules = [
+      TestModule::class,
+      TestLogReportingModule::class,
+      TestLogStorageModule::class,
+      TestDispatcherModule::class,
+      LogUploadWorkerModule::class,
+      TestLogUploaderModule::class
```

Comment:
Ditto for these modules (prefer tightly formatting them).

Reply:
Done.